lib/sda/context.go

```diff
@@ -1,16 +1,17 @@
 package sda
 
-import (
-	"github.com/dedis/cothority/lib/network"
-)
+import "github.com/dedis/cothority/lib/network"
 
-// Context is the interface that is given to an Service
+// Context is the interface that is given to a Service
 type Context interface {
-	NewTreeNodeInstance(*Tree, *TreeNode) *TreeNodeInstance
+	NewTreeNodeInstance(*Tree, *TreeNode, string) *TreeNodeInstance
 	RegisterProtocolInstance(ProtocolInstance) error
 	SendRaw(*network.Entity, interface{}) error
+	CreateProtocolService(*Tree, string) (ProtocolInstance, error)
+	CreateProtocolSDA(*Tree, string) (ProtocolInstance, error)
```

Comment:
(Optional) Maybe having a short comment explaining the purpose and differences of `CreateProtocolService` and `CreateProtocolSDA` could be helpful